core/heap.c

```diff
@@ -1401,6 +1401,9 @@ vmm_heap_init()
     if (DYNAMO_OPTION(vm_reserve)) {
         vmm_heap_unit_init(&heapmgt->vmheap, DYNAMO_OPTION(vm_size));
     }
+    dynamo_vm_areas_lock();
+    heap_exiting = false;
```

Comment:
Right.  (Re: init on ARM (now unrelated to the code): should be fine with just one barrier at the very end.)